js/main.js

```diff
@@ -0,0 +1,119 @@
+(function() {
+
+    function loadPics() {
+        //supplied values from flickr
+        var api_key = '74e47a159e15cbcb6139ba9c9df64c13';
+        var secret = 'c73c5dcbac8e4a96';
+        var method = "flickr.photos.search";
+        var user_id = "138698049@N03";
+        var format = '&format=json';
+        var url = "https://api.flickr.com/services/rest"+"?method="+method+"&api_key="+api_key+"&user_id="+user_id+format+'&nojsoncallback=1';
+
+        var xmlhttp = new XMLHttpRequest();
+
+        xmlhttp.onreadystatechange = function () {
+            var farm, server, id, title, secret, picUrl, list, img, items;
+
+            if (xmlhttp.readyState == 4 && xmlhttp.status == 200) {
+                //make an empty array
+                var pics = [];
+
+                //parse request to json
+                var myResp = JSON.parse(xmlhttp.response);
+                //console.log(myResp);
+
+                for(var i=0; i< myResp.photos.photo.length; i++){
+                    farm = myResp.photos.photo[i].farm;
+                    server = myResp.photos.photo[i].server;
+                    id = myResp.photos.photo[i].id;
+                    title = myResp.photos.photo[i].title;
+                    secret = myResp.photos.photo[i].secret;
+                    picUrl = "https://farm"+farm+".staticflickr.com/"+server+"/"+id+"_"+secret+"_z.jpg";
+
+                    //add each picture url to the pics array
+                    pics.push(picUrl);
+                }
+
+                for(var i=0; i<pics.length; i++){
```

Comment:
Is the `pics.length` the same as `myResp.photos.photo.length`? If so, try to combine into a single loop, with maybe functions handling the dirty work outside of the loop block. Similar to:

```javascript
for(var i = 0; i < array.length; i += 1){
  parsePhotoData(array[i]);
  addToDOM(picUrl);
}
```